### PR TITLE
Big cleanup/refactor

### PR DESCRIPTION
**Pull request by psiemens**
I drank some coffee this afternoon and fell down the refactor rabbit hole. Here's what I changed (everything is backend):

- `post_create` and `post_update` signals are triggered every time a model is created/updated via the API. This was achieved by subclassing `rest_framework.viewsets.ModelViewSet` in `api/mixins.py`
- `actions` are now created by listening to `post_create` and `post_update` signals
- I removed a bunch of fields from the `Publishable` and `Article` models that we don't need anymore (`status`, `images`, `scripts`, `stylesheets`, `snippets`, `est_reading_time`)
- I made it easier to hide certain fields to unauthenticated API requests by subclassing `rest_framework.serializers.HyperlinkedModelSerializer` in `api/mixins.py`
- Various syntax improvements in `ArticleSerializer` and `PageSerializer`
- Removed some old tests that were kicking around
- Removed the `frontpage` endpoint because that's Ubyssey-specific functionality
- Removed the `Comment` model and endpoint (half-baked, not a feature we're shipping anytime soon)
- Removed the `trending` endpoint (it was a dummy endpoint I added a long time ago)

Note: to get this running, you'll have to migrate the database:

```bash
python manage.py makemigrations core
python manage.py makemigrations content
python manage.py migrate core
python manage.py migrate content
```

(Sorry)

### CODE REVIEW

File: dispatch/apps/api/mixins.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    def perform_update(self, serializer):
+        """Override perform_update to send post_update signal"""
+        instance = serializer.save()
+        post_update.send(sender=self.model, instance=instance, user=self.request.user)
+
+class DispatchModelSerializer(HyperlinkedModelSerializer):
+
+    def __init__(self, *args, **kwargs):
+        """Override default constructor to call hide_authenticated_fields"""
```
Comment (blx) [inline dispatch/apps/api/mixins.py]: i think some of these docstrings might make more sense as comments -- they're commenting on why these overrides exist, not so much public docs -- at least phrased how they are currently.

File: dispatch/apps/api/mixins.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        super(DispatchModelSerializer, self).__init__(*args, **kwargs)
+
+        self.hide_authenticated_fields()
+
+    def hide_authenticated_fields(self):
+        """Hide authenicated_fields if user is not authenticated"""
```
Comment (blx) [inline dispatch/apps/api/mixins.py]: `authenicated_` -> `authenticated_`

File: dispatch/apps/api/views.py [hunk 3]
```diff
@@ -242,28 +173,27 @@ class PageViewSet(viewsets.ModelViewSet):
 
     def get_queryset(self):
         """
-        Optionally restricts the returned articles by filtering
-        against a `topic` query parameter in the URL.
+        Only display unpublished content to authenicated users, filter by query parameter if present.
```
Comment (blx) [inline dispatch/apps/api/views.py]: `authenicated` -> `authenticated`

File: dispatch/apps/api/views.py [hunk 4]
```diff
@@ -664,80 +567,16 @@ def list_actions(self, request):
...
-#
-#         data = {
-#             'user': user_serializer.data,
-#             'token': unicode(token)
-#         }
-#         return Response(data, status=status.HTTP_201_CREATED)
-#     except:
-#         return Response(request.data, status=status.HTTP_400_BAD_REQUEST)
-
-
```
Comment (blx) [inline dispatch/apps/api/views.py:775]: 👷  🚧

**Review by blx — COMMENTED** [review]
a few lil things, functionality looks good though! 💃

**Comment by psiemens** [comment]
Sweet, thanks for the review @blx :metal:

**Pull request merged**